pkg/hubble/parser/threefour/parser.go

```diff
@@ -669,8 +669,10 @@ func decodeNetworkInterface(tn *monitor.TraceNotify, dbg *monitor.DebugCapture)
 	}
 }
 
-func decodeProxyPort(dbg *monitor.DebugCapture) uint32 {
-	if dbg != nil {
+func decodeProxyPort(dbg *monitor.DebugCapture, tn *monitor.TraceNotify) uint32 {
+	if tn != nil {
```

Comment:
This needs to also check if `tn.ObsPoint == monitorAPI.TraceToProxy`. Usually, `DstId` refers to the destination endpoint id, only in the ToProxy case will it contain the proxy port.

Reply:
thanks these suggestions! i will update commit:)